celery_worker/worker.py

```
@@ -1,59 +1,125 @@
+            status = "failed"
+            answer = None
+            processed_path = None
+            error_msg = result.get("error", "Unknown agent error")
+
+    # Issue 5: Retry on HTTP 502/503/504
+    except requests.exceptions.HTTPError as http_err:
+        # Retry on 502/503/504 gateway errors
+        if response.status_code in (502, 503, 504):
+            logger.warning(f"Agent returned {response.status_code}, retrying... Response: {response.text}")
+            client.close()
+            raise self.retry(exc=http_err, countdown=30)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)